### DISCUSSION TITLE
Any guidance on adding Sparkle.framework to mac os x app ?

### DISCUSSION THREAD

**Discussion by magbicaleman** [pyinstaller/pyinstaller#7755] [Help]
I'm trying to add https://github.com/sparkle-project/Sparkle/ Sparkle.framework, in to the `Contents/Frameworks` folder using the `datas` array in the spec file, Sparkle.framework is added to the `Contents/Resources` folder. If I try to add it through the binaries array in the spec file I get an error

```
ValueError: Unknown Mach-O header: 0x2f2f0a2f in <_io.BufferedReader name='...etc/Sparkle.framework/Versions/B/Headers/SPUDownloadData.h'>
```

What's the correct approach, or is Sparkle not currently supported?

**Comment by rokm** [comment]
Embedded .framework bundles are currently not properly supported - you'll need to use `develop`, which has #7619 merged.
